widget/richtext_test.go

```diff
@@ -867,6 +867,10 @@ func TestText_findSpaceIndex(t *testing.T) {
 			text: "ww wwww www wwwww",
 			want: 11,
 		},
+		"space beginning": {
+			text: " ww",
+			want: 2,
```

Comment:
It doesn't seem right - the space should be found at position 0, should it not?

Reply:
Hello @andydotxyz. To my understanding, findSpaceIndex is basically a way to check where words are being separated. A space at the beginning of a word is not really separating anything so it would not make sense to treat it as a separator. I agree that as it is now, it's not super clear. I could try to move the change to the `lineBounds` function and check for the zero value there and leave findSpaceIndex to do its things

Reply:
The same could be said about the `fallback` return in findSpaceIndex. Returning the fallback value does not really make it clear if it found a space at lets say return value X or it found none.

Reply:
Actually the above statement I stroke through is correct
If you have "text", you get an index 3 (fallback)
If you have "text " you get again index 3, not 4

Reply:
I was just going by the comments of the method, about returning the last space on a string. Does it not fix the bug if, for leading space, you return 0?
If not then this seems like a good fix - perhaps the method doc just needs a little updating.

Reply:
Unfortunately returning 0 due to leading spaces is what seems to trigger the crash. I'll update the comment tomorrow and let you know, for further approvals

Reply:
> I was just going by the comments of the method

apologies, by method, you mean the function `findSpaceIndex` right?